atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/atomic/ConsensusForgettingStore.java

```diff
@@ -56,19 +57,23 @@ default void mark(Set<Cell> cells) {
      *   another update operation has failed in the past non-deterministically
      *   3. The semantics of the update operation depend on the underlying implementation.
      */
-    void atomicUpdate(Cell cell, byte[] value) throws KeyAlreadyExistsException;
+    ListenableFuture<Void> atomicUpdate(Cell cell, byte[] value) throws KeyAlreadyExistsException;
 
```

Comment:
Actually wondering that using the exception to control logic is not such a great idea. We might benefit from returning a Future<AtomicUpdateResult>